client-tmp/angular/src/app/directive/app-lazy-load.directive.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  Directive,
+  ViewContainerRef,
+  Input,
+  OnInit,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core';
+import { layoutComponentImportsByName } from '../app.config';
+@Directive({
+  selector: '[appLayoutLazyLoad]',
+})
+export class LayoutLazyLoadDirective implements OnInit, OnChanges {
+  @Input('appLayoutLazyLoad') componentName?: string;
+
+  constructor(private viewContainerRef: ViewContainerRef) {}
+
+  ngOnInit() {
+    this.loadComponent();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['componentName']) {
```

Comment:
do you want to check if old and new values are the different here?

Reply:
that's right, thanks for pointing it out
I corrected the condition (by the way, I also changed it in the integration examples)
👍